init.go

```diff
@@ -40,3 +48,44 @@ func init() {
 		panic("libcontainer: container init failed to exec")
 	}
 }
+
+func handleSignals(signals chan os.Signal) {
+	go func() {
+		for {
+			select {
+			case s := <-signals:
+				logrus.Debugf("received signal %v ", s)
+				switch s {
+				case unix.SIGUSR1:
+					dumpStacks(true)
+				default:
+				}
+			}
+		}
+	}()
+}
+
+func dumpStacks(writeToFile bool) {
+	var (
+		buf       []byte
+		stackSize int
+	)
+	bufferLen := 16384
+	for stackSize == len(buf) {
+		buf = make([]byte, bufferLen)
+		stackSize = runtime.Stack(buf, true)
+		bufferLen *= 2
+	}
+	buf = buf[:stackSize]
+
+	if writeToFile {
+		// Also write to file to aid gathering diagnostics
+		name := filepath.Join(os.TempDir(), fmt.Sprintf("runc.%d.stacks.log", os.Getpid()))
+		f, err := os.Create(name)
+		if err != nil {
+			return
+		}
+		defer f.Close()
+		_, _ = f.WriteString(string(buf))
+	}
```

Comment:
There are several ways I can imagine this leading to a vulnerability (a container process would be able to trigger this while runc is joining -- and the non-dumpable bit doesn't help here -- thus causing a more privileged write to a predictable path which could be a symlink or magic-link pointing to who knows where).

I'd prefer we not have this "write to a random file" feature at all. This information should just be output to stderr.

Reply:
If I print these messages to  stderr, how can I get this messages if I don't save these message to a file.

Reply:
If you use `--log` with runc you can pick a file to write to, and in Docker and containerd, runc logs are piped through to their logs so you should be able to access it with `--debug` (or something similar). Another huge benefit of logging (through `logrus` at least) is that if the runc process inside the container gets stuck you'll get the stack information in the host (rather than dumping it somewhere in the container).

Reply:
I print stack into log.json